src/Bicep.Core/Registry/AzureContainerRegistryManager.cs

```diff
@@ -153,13 +153,10 @@ private static async Task<OciArtifactResult> DownloadManifestAsync(IOciArtifactR
             ValidateManifestResponse(manifestResponse);
 
             var deserializedManifest = OciManifest.FromBinaryData(manifestResponse.Value.Manifest) ?? throw new InvalidOperationException("the manifest is not a valid OCI manifest");
-            var layers = new List<(string MediaType, BinaryData Data)>();
-            foreach (var layer in deserializedManifest.Layers)
-            {
-                layers.Add((layer.MediaType, await PullLayerAsync(client, layer)));
-            }
+            var layerTasks = deserializedManifest.Layers
+                .Select(async layer => new OciArtifactLayer(layer.Digest, layer.MediaType, await PullLayerAsync(client, layer)));
 
-            return new(manifestResponse.Value.Manifest, manifestResponse.Value.Digest, layers.ToImmutableArray());
+            return new(manifestResponse.Value.Manifest, manifestResponse.Value.Digest, await Task.WhenAll(layerTasks));
```

Comment:
Is this a deliberate decision to switch from executing sequentially -> in parallel? Does the parallelism need a limit?

Reply:
It was deliberate, I thought to optimize the speed since this is invoked in the lang server reconcile loop and I/O is slow . I am limiting the parallelism to 5 now, hopefully that is enough